src/mono/wasm/debugger/DebuggerTestSuite/ExceptionTests.cs

```diff
@@ -0,0 +1,254 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using Microsoft.WebAssembly.Diagnostics;
+using Xunit;
+
+namespace DebuggerTests
+{
+
+    public class ExceptionTests : DebuggerTestBase
+    {
+        [Fact]
+        public async Task ExceptionTestAll()
+        {
+            var insp = new Inspector();
+            //Collect events
+            var scripts = SubscribeToScripts(insp);
+            int line = 15;
+            int col = 20;
+            string entry_method_name = "[debugger-test] DebuggerTests.ExceptionTestsClass:TestExceptions";
+
+            await Ready();
+            await insp.Ready(async(cli, token) =>
+            {
+                ctx = new DebugTestContext(cli, insp, token, scripts);
+                var debugger_test_loc = "dotnet://debugger-test.dll/debugger-exception-test.cs";
+
+                await SetPauseOnException("all");
+
+                var eval_expr = "window.setTimeout(function() { invoke_static_method (" +
+                    $"'{entry_method_name}'" +
+                    "); }, 1);";
+
+                JObject pause_location;
+
+                //skipping js caught exception
+                await EvaluateAndCheck(eval_expr,  null, 0, 0, null);
+                do 
+                {
+                    pause_location = await SendCommandAndCheck(JObject.FromObject(new { }), "Debugger.resume",  null, 0, 0, null);
+                } while (pause_location["callFrames"] ? [0] ? ["functionName"]?.Value<string>() != "run");
```

Comment:
I think we should check that we pause only for exceptions, and wait for the first managed one. And then check if that is on our expected method. How about something like - https://gist.github.com/623893730a89b2eb3bc652fbf6a1fa31 ?

Reply:
Done! :)